Kernel/Interrupts/InterruptManagement.cpp

```diff
@@ -0,0 +1,227 @@
+#include <AK/FixedArray.h>
+#include <Kernel/ACPI/MultiProcessorParser.h>
+#include <Kernel/Arch/i386/CPU.h>
+#include <Kernel/Interrupts/IOAPIC.h>
+#include <Kernel/Interrupts/InterruptManagement.h>
+#include <Kernel/Interrupts/PIC.h>
+#include <Kernel/VM/MemoryManager.h>
+
+#define PCAT_COMPAT_FLAG 0x1
+namespace Kernel {
+static InterruptManagement* s_interrupt_management;
+static bool initialized;
+
+InterruptManagement& InterruptManagement::the()
+{
+    if (!initialized && s_interrupt_management == nullptr)
+        s_interrupt_management = new InterruptManagement(false);
+    return *s_interrupt_management;
+}
+
+AdvancedInterruptManagement& AdvancedInterruptManagement::the()
+{
+    ASSERT(s_interrupt_management != nullptr);
+    return *(AdvancedInterruptManagement*)s_interrupt_management;
+}
+
+void InterruptManagement::initialize()
+{
+    if (!initialized)
+        s_interrupt_management = new InterruptManagement(true);
+    initialized = true;
+}
```

Comment:
It's very hard to understand the lifetime model of this kind of object.
Instead, I suggest getting rid of automatic instantiation and creating objects like this explicitly on the stack in `init.cpp`
That way, we can see what objects are being created during boot and when.

Reply:
I'll work on a solution, I agree it's pretty much complicated to see how it is initialized and created.